src/test/java/org/javarosa/core/storage/IndexedStorageUtilityTests.java

```diff
@@ -131,22 +128,32 @@ public void testBulkMetaMatching() {
         Assert.assertEquals("Failed index match [brand,style][nike,mens]", getIdsFromModels(tenSizesOfMensNikes), getIdsFromModels(matchedRecords.toArray(new Shoe[]{})));
     }
 
+    @Test
+    public void testReadingAllEntries() {
+        writeBulkSets();
+        List<Integer> matches = storage.getIDsForValues(new String[]{}, new String[]{});
```

Comment:
Style nit: You can also instantiate this with `new String[0]` which might make the intent slightly clearer